src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/OutlineAccessibility.m

```diff
@@ -111,4 +111,32 @@ - (void)invalidateSelectionCache
     selectedRowCacheValid = NO;
 }
 
+- (void)postSelectionChanged
+{
+    AWT_ASSERT_APPKIT_THREAD;
+    [self invalidateSelectionCache];
+    [super postSelectionChanged];
+}
+
+- (void)postTreeNodeCollapsed
+{
+    AWT_ASSERT_APPKIT_THREAD;
+    [self invalidateCache];
+    [super postTreeNodeCollapsed];
+}
+
+- (void)postTreeNodeExpanded
+{
+    AWT_ASSERT_APPKIT_THREAD;
+    [self invalidateCache];
+    [super postTreeNodeExpanded];
+}
+
+- (void)postSelectedCellsChanged
```

Comment:
Does OutlineRowAccessibility use CellAccessibility?

Reply:
It does not - i am just covering all methods that can be potentially called on the table-like controls. It does no harm to override method that will never be called. But it will cause some discrepancies in the behavior if some custom control will call this notification and we fail to update the selection cache accordingly.